### PR TITLE
[encryption] fix move versions between storages

### LINKED ISSUE #13668: Cannot download encrypted file version after moving to external storage
### Steps to reproduce
1. Enable encryption, external storage and files locking app
2. Log out then log in again
3. Mount a SFTP external storage to "/sftp"
4. Create a text file in the home root: "test.txt"
5. Edit the text file several times to create versions
6. Check the versions dropdown to have at least 3 versions
7. Check that an older version can be downloaded
8. Now move "test.txt" to "/sftp"
9. Check the versions dropdown
10. Download a version

### Expected result
Versions can be downloaded, preview looks alright.

### Actual result
Version file is still encrypted after download. The text preview thumbnail shows an encrypted content.
I haven't tried without the locking app, but suspect that it might happen without it too.

### Versions
Happens on OC 8 master (f8b1fde2c06575de79035f6690bda8a24c9802e9)
and on OC 7 stable7 (80560e70b7133739f87cf6a03acb55169fb2462f)

### Logs
Log (with the files_locking entries removed as it is too long)
```
{"reqId":"54c642d56a206","app":"PHP","message":"ob_end_clean(): failed to delete buffer. No buffer to delete at \/srv\/www\/htdocs\/owncloud\/lib\/private\/files\/view.php#269","level":0,"time":"2015-01-26T13:36:21+00:00","method":"GET","url":"\/owncloud\/index.php\/apps\/files_versions\/download.php?file=%2Fsftp%2Fv2.txt&revision=1422279013"}
```

@schiesbn

### PR DESCRIPTION
**Pull request by schiessle**
fix https://github.com/owncloud/core/issues/13668

cc @PVince81 please review, thanks!

### CODE REVIEW

**Comment by PVince81** [comment]
The original test steps work.

However, moving the file from a system-wide mount point (SFTP) to another personal mount point (SFTP) corrupts the versions.

**Comment by PVince81** [comment]
(moved directly with WebDAV, as the web UI doesn't let you move to parallel folders)

**Comment by schiesbn** [comment]
This sound like this issue: https://github.com/owncloud/core/issues/16373

For your test case you probably need to combine this pr with this one: https://github.com/owncloud/core/pull/16526

**Comment by PVince81** [comment]
I tried merging https://github.com/owncloud/core/pull/16526 into this PR but it didn't solve the cross-storage issue.

**Comment by schiesbn** [comment]
Fixed, please try again. Thanks!

**Comment by PVince81** [comment]
Tested by @jvillafanez https://github.com/owncloud/core/issues/16356#issuecomment-104611003 , counting as :+1:

**Comment by PVince81** [comment]
Works now :+1: (without additional PRs/merges)

**Comment by PVince81** [comment]
Needs a second reviewer @th3fallen @DeepDiver1975 @icewind1991

**Comment by cmonteroluque** [comment]
@th3fallen any chance you can be the second reviewer?

File: lib/private/files/storage/wrapper/encryption.php [hunk 1]
```diff
@@ -289,34 +297,32 @@ public function isReadable($path) {
 	 */
 	public function copy($path1, $path2) {
 
-		$fullPath1 = $this->getFullPath($path1);
-		$fullPath2 = $this->getFullPath($path2);
+		$source = $this->getFullPath($path1);
+		$target = $this->getFullPath($path2);
```
Comment (th3fallen) [inline lib/private/files/storage/wrapper/encryption.php]: THANK YOU god those variables where so confusing....

**Comment by th3fallen** [comment]
works here :+1:

**Comment by davitol** [comment]
:+1:

**Pull request merged**